android/src/main/java/com/reactnativemembrane/MembraneWebRTC.kt

```diff
@@ -35,791 +35,810 @@ import org.webrtc.Logging
+  private fun rtcInboundStatsToRNMap(stats: RTCInboundStats): Map<String, Any?> {
+    val res = mutableMapOf<String, Any?>()
+    res["kind"] = stats.kind
+    res["jitter"] = stats.jitter ?: 0.0
+    res["packetsLost"] = stats.packetsLost ?: 0
+    res["packetsReceived"] = stats.packetsReceived?.toInt() ?: 0
+    res["bytesReceived"] = stats.bytesReceived?.toInt() ?: 0
+    res["framesReceived"] = stats.framesReceived ?: 0
+    res["frameWidth"] = stats.frameWidth?.toInt() ?: 0
+    res["frameHeight"] = stats.frameHeight?.toInt() ?: 0
+    res["framesPerSecond"] = stats.framesPerSecond ?: 0.0
+    res["framesDropped"] = stats.framesDropped?.toInt() ?: 0
+
+    return res
+  }
+
+  fun getStatistics(): MutableMap<String, Map<String, Any?>> {
+    ensureCreated()
+    val newMap = mutableMapOf<String, Map<String, Any?>>()
+    membraneRTC?.getStats()?.forEach { entry ->
+      newMap[entry.key] = if (entry.value is RTCInboundStats) rtcInboundStatsToRNMap(
+        entry.value as RTCInboundStats
+      ) else rtcOutboundStatsToRNMap(entry.value as RTCOutboundStats)
+    }
+    return newMap
+  }
+
+  private fun startScreencast(mediaProjectionPermission: Intent) {
+    localScreencastId = UUID.randomUUID().toString()
+    val videoParameters = getScreencastVideoParameters()
+    val screencastTrack = membraneRTC?.createScreencastTrack(
+      mediaProjectionPermission, videoParameters, screencastMetadata
+    ) ?: throw CodedException("Failed to Create ScreenCast Track")
+    localScreencastTrack = screencastTrack
+    addTrackToLocalEndpoint(screencastTrack, screencastMetadata, videoParameters.simulcastConfig)
+    setScreencastTrackState(true)
+
+    screencastPromise?.resolve(isScreencastOn)
+  }
+
+  private fun getScreencastVideoParameters(): VideoParameters {
+    val videoParameters = when (screencastQuality) {
+      "VGA" -> VideoParameters.presetScreenShareVGA
+      "HD5" -> VideoParameters.presetScreenShareHD5
+      "HD15" -> VideoParameters.presetScreenShareHD15
+      "FHD15" -> VideoParameters.presetScreenShareFHD15
+      "FHD30" -> VideoParameters.presetScreenShareFHD30
+      else -> VideoParameters.presetScreenShareHD15
+    }
+    val dimensions = videoParameters.dimensions.flip()
+    return videoParameters.copy(
+      dimensions = dimensions,
+      simulcastConfig = screencastSimulcastConfig,
+      maxBitrate = screencastMaxBandwidth,
+    )
+  }
+
+  private fun setScreencastTrackState(isEnabled: Boolean) {
+    isScreencastOn = isEnabled
+    val eventName = EmitableEvents.IsScreencastOn
+    emitEvent(eventName, mapOf(eventName to isEnabled))
+    emitEndpoints()
+  }
+
+  private fun stopScreencast(calledFromToggleScreencast: Boolean = true) {
+    ensureScreencastTrack()
+    localScreencastTrack?.let {
+      removeTrackFromLocalEndpoint(it)
+      localScreencastTrack = null
+    }
+
+    setScreencastTrackState(false)
+    if (calledFromToggleScreencast) {
+      screencastPromise?.resolve(isScreencastOn)
+    }
```

Comment:
Why we need this condition? Does resolving promise in disconnect cause some issues?